### ISSUE TITLE
Server Lag on PMC--During PvP and otherwise (Multiple People) 

### ISSUE THREAD

**Issue by CreatingWithKass** [PirateCraft/dev#134]
**Describe the bug** *(A clear and concise description of what the bug is)*:
This has been an issue for awhile, one I've put off making a report on until I had enough evidence to back up claims (from multiple sources). Basically, PMC gets extremely laggy sometimes, especially during PvP whether that is between one on one fights or multiple people participating at once. It has been tested by multiple different players on every version of Minecraft, in different settings (/warp south, arena, etc), and on laptops to desktop computers; the problem persists. I have been asked to report it and see if there's anything that can be done. 

**To Reproduce** *(Steps to reproduce the behavior)*:
Try to fight on PMC. 

**Time and Date** *(If applicable, add the time (in GMT+1) and date (DD.MM.YYYY) of the issue)*:
05/05/2021 

**Screenshots/Videos (Delete if not needed)** *(If applicable, add screenshots/videos to help explain your problem)** 
1. https://www.youtube.com/watch?v=gBtLBj6Ljrs
    - Timestamps: **2:52 is a MAJOR one, where a player dies due to server lag**, 0:19 , 0:30
 2. **This final video has some very good clips of pot lag, food lag and more. Please take a look at it.** 
    - https://youtu.be/Kq6JT5d_bAs
3. https://youtu.be/NrtVP7w57kU
4. https://www.youtube.com/watch?v=m2f_BQxFjA0 
5. https://www.youtube.com/watch?v=7Ane-JEiZcA
    - Timestamps for this vid: 0:07 , 1:03 , 1:07 - 1:16 , 1:30+, 2:18 , **2:48 - 2:50 are pretty major spikes** 
6. https://www.youtube.com/watch?v=t979skn93nI
    - Timestamps: 0:10 , 0:39 , 1:29 , 2:28 

**Additional context (Delete if this isn't needed)** *(Add any other context about the problem here)*:
Many believe it is due to the location of the server's host, however I do not run this server and neither do they, so it's just speculation. You are the brains behind the server so I figured creating one big report with multiple videos could help in the investigation.

**Comment by TomLewis** [comment]
Literally impossible to diagnose without a way to reproduce, so I'm going to take an educated guess;

Did this start around the same time I added that `/recount` plugin to track who did damage to you and how you died? (If someone can verify when this started we can look over the changelogs!)

This would be the only thing I can think of that would create lag as it would be temporarily recording each and every piece of damage done to a person.

Since this is going to have to be trial and error, I will start by removing that plugin entirely and everyone that made videos here can re-test and get back to me if that was the issue.

**Comment by CreatingWithKass** [comment]
> Literally impossible to diagnose without a way to reproduce, so I'm going to take an educated guess;
> 
> Did this start around the same time I added that `/recount` plugin to track who did damage to you and how you died? (If someone can verify when this started we can look over the changelogs!)
> 
> This would be the only thing I can think of that would create lag as it would be temporarily recording each and every piece of damage done to a person.
> 
> Since this is going to have to be trial and error, I will start by removing that plugin entirely and everyone that made videos here can re-test and get back to me if that was the issue.

According to the people making these reports, the lag has been happening for at least a year. I’m not sure when you added the recount plug-in, but we will test after the server restart and get back to you. I’ll make sure everyone starts recording the number of people online, location, version, etc as we try to narrow this down. 

I also talked to Encode and he will need some information if he’s to offer us any advice if you’re interested. We can talk it over in a staff call if I can manage to get off work for it sometime soon!

**Comment by TomLewis** [comment]
Yeah I too will need some information haha, way too vauge of a report with no way to reproduce and without people giving a timeline to when it started for me to figure out what was changed, its entirely a guessing game.


Get people to test on the test server too, I can remove plugins from that much easier.

**Comment by CreatingWithKass** [comment]
Adding more possible leads: 
_______________
Armando: 
Issue: there are 2 anticheats on the server and that might be causing some lag issues, one is a white message that pops up when you're kicked for being on a horse without a saddle (this particular kick has no rejoin cooldown), and the other one is the normal anticheat that kicks you for, example: breaking blocks too fast, having good aim, using the new speed 4 brews, etc. with a rejoin cooldown.
_______________

Not sure if this helps at all but it is good to document.

**Comment by TomLewis** [comment]
> there are 2 anticheats on the server and that might be causing some lag issues
That's not true, never have we had 2 AC's.
